src/frontend/komponenter/Fagsak/Vilkårsvurdering/GeneriskVilkår/Vilkår/BarnetsAlder/BarnetsAlderContext.ts

```diff
@@ -105,8 +110,47 @@ export const useBarnetsAlder = (
                 erEksplisittAvslagPåSøknad: erEksplisittAvslagPåSøknad.verdi,
             },
         }),
+        adopsjonsdato: useFelt<Date | undefined>({
+            verdi: undefined,
+            valideringsfunksjon: felt => {
+                if (!felt.verdi || !isValid(felt.verdi)) {
+                    return feil(felt, 'Adopsjonsdato må fylles ut når adopsjon er valgt');
+                } else {
+                    return ok(felt);
+                }
+            },
+            avhengigheter: { utdypendeVilkårsvurdering: utdypendeVilkårsvurdering.verdi },
+            skalFeltetVises: (avhengigheter: Avhengigheter) =>
+                avhengigheter?.utdypendeVilkårsvurdering.includes(
+                    UtdypendeVilkårsvurderingGenerell.ADOPSJON
+                ),
+        }),
     };
 
+    const [forrigeAdopsjonsdato, settForrigeAdopsjonsdato] = useState<IsoDatoString | undefined>();
+
+    const settAdopsjonsdatoTilDefault = () => {
+        felter.adopsjonsdato.validerOgSettFelt(
+            person.adopsjonsdato ? new Date(person.adopsjonsdato) : undefined
+        );
+    };
+
+    if (person.adopsjonsdato !== forrigeAdopsjonsdato) {
+        settForrigeAdopsjonsdato(person.adopsjonsdato);
+        settAdopsjonsdatoTilDefault();
+    }
+
+    useEffect(() => {
+        if (
+            felter.utdypendeVilkårsvurdering.verdi.includes(
+                UtdypendeVilkårsvurderingGenerell.ADOPSJON
+            ) &&
+            !felter.adopsjonsdato.verdi
+        ) {
+            settAdopsjonsdatoTilDefault();
+        }
+    }, [felter.utdypendeVilkårsvurdering.verdi]);
```

Comment:
Dette sørger for å oppdatere adopsjonsdato-feltet med lagret verdi fra backend hvis man går fra å ikke ha valgt adopsjon i utdypende vilkårsvurdering til å gjøre det

Reply:
Dette er for å støtte tilfeller hvor vi får to perioder på vilkåret. Da skal man bare legge inn dato på første perioden også skal det dukke opp på andre perioden av seg selv ✨

Reply:
Det er nok noe jeg ikke ser her, så du får ha meg unnskyldt 🙈. Men vil ikke feltet oppdatere seg når adopsjonsdato oppdateres i behandlingen dersom man setter verdien til feltet til å alltid være `person.adopsjonsdato ? new Date(person.adopsjonsdato) : undefined`?